packages/salesforcedx-lwc-language-server/src/componentIndexer.ts

```diff
@@ -0,0 +1,307 @@
+                return this.tags.get(name) ?? this.tags.get(snakeCase(name)) ?? null;
+            }
+            if (delimiter === LWC_DELIMITER) {
+                return this.tags.get(name) ?? this.tags.get(camelCase(name)) ?? null;
+            }
+            return this.tags.get(query) ?? null;
+        } catch {
+            return null;
+        }
+    }
+
+    public findTagByURI(uri: string): Tag | null {
+        const uriText = uri.replace('.html', '.js');
+        return Array.from(this.tags.values()).find((tag) => getTagUri(tag) === uriText) ?? null;
+    }
+
+    public async loadTagsFromIndex(): Promise<void> {
+        try {
+            const indexPath: string = path.join(this.workspaceRoot, CUSTOM_COMPONENT_INDEX_FILE);
+            const uri = `file://${indexPath}`;
+
+            if (this.fileSystemProvider.fileExists(uri)) {
+                const content = this.fileSystemProvider.getFileContent(uri);
+                if (content) {
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                    const index: TagAttrs[] = JSON.parse(content);
+                    for (const data of index) {
+                        const info = await createTag(data);
+                        this.tags.set(getTagName(info), info);
+                    }
+                }
+            }
+        } catch (err) {
+            console.error(err);
+        }
+    }
+
+    public persistCustomComponents(): void {
+        const indexJsonString = JSON.stringify(this.getCustomData());
+
+        // Store the component index data for the client to process
+        this.fileSystemProvider.updateFileContent('lwc:componentIndex', indexJsonString);
+    }
+
+    public async insertSfdxTsConfigPath(filePaths: string[]): Promise<void> {
+        const sfdxTsConfigPath = normalize(`${this.workspaceRoot}/.sfdx/tsconfig.sfdx.json`);
+        const uri = `file://${sfdxTsConfigPath}`;
+
+        const fileExists = this.fileSystemProvider.fileExists(uri);
+
+        if (fileExists) {
+            try {
+                const sfdxTsConfig: SfdxTsConfig = readJsonSync(sfdxTsConfigPath, this.fileSystemProvider);
+                sfdxTsConfig.compilerOptions = sfdxTsConfig.compilerOptions ?? { paths: {} };
+                sfdxTsConfig.compilerOptions.paths = sfdxTsConfig.compilerOptions.paths ?? {};
+                for (const filePath of filePaths) {
+                    const { dir, name: fileName } = path.parse(filePath);
+                    const componentName = `c/${fileName}`;
+                    const componentFilePath = path.join(dir, fileName);
+                    const tsConfigFilePaths: string[] = (sfdxTsConfig.compilerOptions.paths[componentName] ??= []);
+                    const hasExistingPath = tsConfigFilePaths.includes(componentFilePath);
+                    if (!hasExistingPath) {
+                        tsConfigFilePaths.push(componentFilePath);
+                    }
```

Comment:
why does it matter if you push them to tsConfigFilePaths?  I don't see it used to do/decide anything.

Maybe there's a bug in this?